modules/1_opencv/src/opencv_coins.cpp

```diff
@@ -1,13 +1,60 @@
 #include "opencv_coins.hpp"
-
+#include <opencv2/opencv.hpp>
+#include <math.h>
+#include <algorithm>
 using namespace cv;
 
 unsigned countCoins(const Mat& img) {
-    Mat gray, thresh;
+    Mat gray, thresh, newThresh;
     cvtColor(img, gray, COLOR_BGR2GRAY);
     threshold(gray, thresh, 0, 255, THRESH_BINARY_INV + THRESH_OTSU);
     // Remove unfilled black holes by morphological closing
     morphologyEx(thresh, thresh, MORPH_CLOSE, Mat::ones(3, 3, CV_8U), Point(-1, -1), 3);
+        
+    //imshow("sad", thresh);
+    //waitKey(9000);
+    
+    dilate(thresh, newThresh, Mat::ones(3, 3, CV_8U), Point(-1, -1), 3);
+    //imshow("22", newThresh);
+    //waitKey(9000);
+
+    erode(thresh, thresh, Mat::ones(3, 3, CV_8U), Point(-1, -1), 5);
+    distanceTransform(thresh, thresh, DIST_L2, 3);
+    //imshow("33", thresh);
+    //waitKey(9000);
+
+    double minDistance, maxDistance;
+    minMaxLoc(thresh, &minDistance, &maxDistance);
+    threshold(thresh, thresh, maxDistance * 0.7, 255, THRESH_BINARY);
+    //imshow("55", thresh);
+    //waitKey(9000);
+    
+    thresh.convertTo(thresh, CV_8U, 1, 0);
+
+    std::vector<std::vector<Point>> contoursVector;
+    findContours(thresh, contoursVector, RETR_EXTERNAL, CHAIN_APPROX_NONE);
+
+    int coinsSum(0);
+    double R(0);
+    std::vector<double> area;
+    for (int i(0); i < contoursVector.size(); i++) {
+        area.push_back(contourArea(contoursVector[i]));
+        sort(area.begin(), area.end());
+    }
+    double minRadius2 = sqrt(area[10] / CV_PI);
```

Comment:
Please try to avoid hardcoded numbers. What if we had less coins?

Reply:
I thought I can use the number of coins to solve

Before "return" I wrote the second solution option without using the information of amount.

Reply:
Please remove commented code